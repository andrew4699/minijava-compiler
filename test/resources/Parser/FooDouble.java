class Foo {
  /***/
  public/*
     *x
     */

  /***/
  static/*
     *
     */ void main(String[] args) {
    System.out.println(5e25 + 5e-33 * 4e99 * 3 - 1.);
  }
}

class Bar extends Nothing { public double bar(Baz f, double b) {
    double y;
    y = 1.;
    System.out.println(new Thing().method());
    return 5.3e66;
  }
}
