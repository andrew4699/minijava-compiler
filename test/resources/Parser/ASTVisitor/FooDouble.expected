Program
  MainClass Foo (line 1)
    Print (line 11)
      ((5.0E25 + ((5.0E-33 * 4.0E99) * 3)) - 1.0)
  Class Bar extends Nothing (line 15)
    MethodDecl bar (line 15)
      returns double
      parameters:
        Baz f
        double b
      VarDecl double y (line 16)
      y = 1.0
      Print (line 18)
        new Thing().method()
      Return 5.3E66 (line 19)
